ember-client/shared/src/main/scala/org/http4s/ember/client/EmberConnection.scala

```diff
@@ -42,4 +47,10 @@ private[ember] object EmberConnection {
         EmberConnection(keySocket, release, nextBytes)
       }
     }
+
+  // This implicit conversion only kicks in on JS platform to add this syntax
+  @nowarn("msg=never used")
+  private implicit final class SocketOpsJS[F[_]](val socket: Socket[F]) extends AnyVal {
+    def endOfInput(implicit F: Monad[F]): F[Unit] = F.unit
+  }
```

Comment:
It could, I wasn't sure what's the right thing to do here. Node.js doesn't support this method and so not making it available is the clearest way to indicate that. I'll raise it up again.